x/resource/types/params.go

```diff
@@ -1,146 +1,307 @@
 package types
 
 import (
+	"context"
 	"fmt"
 
 	sdkmath "cosmossdk.io/math"
+	util "github.com/cheqd/cheqd-node/util"
+	didtypes "github.com/cheqd/cheqd-node/x/did/types"
+	oracletypes "github.com/cheqd/cheqd-node/x/oracle/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-// NewParams creates a new FeeParams object with specified parameters
-func NewParams(image, json, defaultFee sdk.Coin, burnFactor sdkmath.LegacyDec) FeeParams {
-	return FeeParams{
-		Image:      image,
-		Json:       json,
-		Default:    defaultFee,
-		BurnFactor: burnFactor,
-	}
-}
-
 // DefaultFeeParams returns default cheqd module tx fee parameters
 func DefaultFeeParams() *FeeParams {
 	return &FeeParams{
+		Image: []didtypes.FeeRange{
+			{
+				Denom:     BaseMinimalDenom,
+				MinAmount: sdkmath.NewInt(20000000000),
+				MaxAmount: util.PtrInt(30000000000),
+			},
+		},
+		Json: []didtypes.FeeRange{
+			{
+				Denom:     BaseMinimalDenom,
+				MinAmount: sdkmath.NewInt(3500000000),
+				MaxAmount: util.PtrInt(60000000000),
+			},
+		},
+		Default: []didtypes.FeeRange{
+			{
+				Denom:     BaseMinimalDenom,
+				MinAmount: sdkmath.NewInt(6000000000),
+				MaxAmount: util.PtrInt(20000000000),
+			},
+		},
+		BurnFactor: sdkmath.LegacyMustNewDecFromStr(DefaultBurnFactor),
+	}
+}
+
+func DefaultLegacyFeeParams() *LegacyFeeParams {
+	return &LegacyFeeParams{
 		Image:      sdk.NewCoin(BaseMinimalDenom, sdkmath.NewInt(DefaultCreateResourceImageFee)),
 		Json:       sdk.NewCoin(BaseMinimalDenom, sdkmath.NewInt(DefaultCreateResourceJSONFee)),
 		Default:    sdk.NewCoin(BaseMinimalDenom, sdkmath.NewInt(DefaultCreateResourceDefaultFee)),
 		BurnFactor: sdkmath.LegacyMustNewDecFromStr(DefaultBurnFactor),
 	}
 }
 
-// ValidateBasic performs basic validation of cheqd module tx fee parameters
-func (tfp *FeeParams) ValidateBasic() error {
-	if !tfp.Image.IsPositive() || tfp.Image.Denom != BaseMinimalDenom {
-		return fmt.Errorf("invalid create resource image tx fee: %s", tfp.Image)
-	}
+// USDRange represents a fee range converted to USD for comparison
+type USDRange struct {
+	MinUSD sdkmath.LegacyDec
+	MaxUSD *sdkmath.LegacyDec // nil means no upper bound
+}
```

Comment:
This is a helper type to express the converted range. Therefore, rename it to capture this intent, avoiding any possible confusion. Both denominators should be defined as integers within `FeeParams`.